x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpLogic } from '../../../shared/http';
+
+import { IndexingStatus } from '../schema/types';
+import { EngineDetails } from './types';
+
+interface EngineValues {
+  dataLoading: boolean;
+  engine: EngineDetails | {};
```

Comment:
This new `| {}` is required to get Typescript passing, and I believe we confirmed that we wanted to do this per a previous PR discussion - but as a heads up it will affect a significant amount of downstream code that uses EngineValues. We'll find out :)